include/envoy/network/connection.h

```diff
@@ -240,13 +240,13 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual const ConnectionSocket::OptionsSharedPtr& socketOptions() const PURE;
 
   /**
-   * Object on which filters can share data on a per-connection basis.
-   * Only one filter can produce a named data object, but it may be
-   * consumed by many other objects.
-   * @return the per-connection state associated with this connection.
+   * The StreamInfo object associated with this connection. This is typically
+   * used for logging purposes. Individual filters may add specific information
+   * via the FilterState object within the StreamInfo object.
+   *
+   * @return StreamInfo object associated with this connection.
    */
-  virtual StreamInfo::FilterState& perConnectionState() PURE;
-  virtual const StreamInfo::FilterState& perConnectionState() const PURE;
+  virtual StreamInfo::StreamInfo& streamInfo() PURE;
```

Comment:
add back the const interface too. otherwise callers with `const Connection&` cannot read streamInfo.

Reply:
Addressed in dc99eb4.

Reply:
Why do we need the const interface? Put another way, I did the non const thing by looking at the HTTP filter interface that was supplying a non const StreamInfo.

Reply:
Some filter have const reference/pointer to Connection (e.g. RBAC Engine match), in those case they only need to read StreamInfo hence you need the const interface too. It is not blocking you get a mutable reference from non-const ones.